sdk/metric/controller/push/push.go

```diff
@@ -73,24 +74,26 @@ var _ Ticker = realTicker{}
 // using the provided integrator, exporter, collection period, and SDK
 // configuration options to configure an SDK with periodic collection.
 // The integrator itself is configured with the aggregation selector policy.
-func New(integrator export.Integrator, exporter export.Exporter, period time.Duration, opts ...Option) *Controller {
+func New(integrator export.Integrator, exporter export.Exporter, configLoaderCh chan struct{},
+	period time.Duration, opts ...Option) *Controller {
 	c := &Config{ErrorHandler: sdk.DefaultErrorHandler}
 	for _, opt := range opts {
 		opt.Apply(c)
 	}
 
 	impl := sdk.NewAccumulator(integrator, sdk.WithErrorHandler(c.ErrorHandler))
 	return &Controller{
-		accumulator:  impl,
-		resource:     c.Resource,
-		uniq:         registry.NewUniqueInstrumentMeterImpl(impl),
-		named:        map[string]metric.Meter{},
-		errorHandler: c.ErrorHandler,
-		integrator:   integrator,
-		exporter:     exporter,
-		ch:           make(chan struct{}),
-		period:       period,
-		clock:        realClock{},
+		accumulator:  	impl,
+		resource:     	c.Resource,
+		uniq:         	registry.NewUniqueInstrumentMeterImpl(impl),
+		named:        	map[string]metric.Meter{},
+		errorHandler: 	c.ErrorHandler,
+		integrator:   	integrator,
+		exporter:     	exporter,
+		ch:           	make(chan bool),
+		period:       	period,
```

Comment:
The initial period should be read from OnInitialConfig()

Reply:
Let me know if I implemented this in the way you were thinking.

Reply:
As part of my implementation, I set the period to a minute for safety reasons if `OnInitialConfig` is never called. Again, let met know if you had a different implementation in mind!